### ISSUE TITLE
Collection renderer passes the same object to override methods for every collection item

### ISSUE THREAD

**Issue by krzysiek1507** [vasilakisfil/SimpleAMS#25]
```ruby
# frozen_string_literal: true

class TestSerializer
  include SimpleAMS::DSL

  adapter SimpleAMS::Adapters::AMS

  attributes :name

  def name
    object.object_id
  end
end
```
Result:
```ruby
SimpleAMS::Renderer::Collection.new([1,2], serializer: TestSerializer).as_json
=> [{:name=>3}, {:name=>3}]
```

Correct result:
```ruby
[{:name=>3}, {:name=>5}]
```

**Comment by krzysiek1507** [comment]
It looks like this code causes the problem: https://github.com/vasilakisfil/SimpleAMS/blob/d2e4732c9afd750df89608712028cfa86588d385/lib/simple_ams/options.rb#L307-L315

**Comment by vasilakisfil** [comment]
Thanks, I ll take a look today, I think the tracked properties are not tracking the case when you have a set a custom method.

**Comment by krzysiek1507** [comment]
It maybe also a problem that [this](https://github.com/vasilakisfil/SimpleAMS/blob/d2e4732c9afd750df89608712028cfa86588d385/lib/simple_ams/document.rb#L125) reuses options instance but [here](https://github.com/vasilakisfil/SimpleAMS/blob/d2e4732c9afd750df89608712028cfa86588d385/lib/simple_ams/options.rb#L21-L28) it doesn't clear `@serializer` and it reuses the instance for the first item.

**Comment by vasilakisfil** [comment]
Fixed in #27. Just update your Gemfile to 0.2.4. Thanks for reporting!

**Comment by krzysiek1507** [comment]
Thanks!
